cli/search/search.go

```diff
@@ -90,3 +93,106 @@ func SearchVersions(cmdCtx *cmdcontext.CmdCtx, program string) error {
 	os.Stdout.Write([]byte("master\n"))
 	return err
 }
+
+func getDirFiles(path string) ([]fs.FileInfo, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	files, err := f.Readdir(0)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	return files, err
+}
+
+// RunCommandAndGetOutputDir returns output of command.
+func RunCommandAndGetOutputDir(program string, dir string, args ...string) (string, error) {
+	cmd := exec.Command(program, args...)
+	cmd.Dir = dir
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+func SearchVersionsLocal(cmdCtx *cmdcontext.CmdCtx, program string) error {
+	var err error
+	cliOpts, err := configure.GetCliOpts(cmdCtx.Cli.ConfigPath)
+	if err != nil {
+		return err
+	}
+	if cliOpts.Repo == nil {
+		cliOpts.Repo = &config.RepoOpts{Install: "", Rocks: ""}
+	}
+	localDir := cliOpts.Repo.Install
+	if localDir == "" {
+		localDir = cmdCtx.Cli.ConfigPath + "/distfiles"
+	}
+
+	localFiles, err := getDirFiles(localDir)
+	if err != nil {
+		return err
+	}
+
+	log.Warn("Available versions of " + program + ":")
+	if program == "tarantool" {
+		if _, err = os.Stat(localDir + "/tarantool"); !os.IsNotExist(err) {
+			versions, err := RunCommandAndGetOutputDir("git", localDir+"/tarantool",
+				"-c", "versionsort.suffix=-",
+				"tag", "--sort="+"v:refname")
+			if err != nil {
+				return err
+			}
+			versionsArray := strings.Split(versions, "\n")
+			for i := 0; i < len(versionsArray); i++ {
+				if isDeprecated(versionsArray[i]) {
+					continue
+				}
+				os.Stdout.Write([]byte(versionsArray[i]))
+				os.Stdout.Write([]byte("\n"))
+			}
+			os.Stdout.Write([]byte("master\n"))
+		}
+	} else if program == "tt" {
+		if _, err = os.Stat(localDir + "/tt"); !os.IsNotExist(err) {
+			versions, err := RunCommandAndGetOutputDir("git", localDir+"/tt", "-c",
+				"versionsort.suffix=-",
+				"tag", "--sort="+"v:refname")
+			if err != nil {
+				return err
+			}
+			os.Stdout.Write([]byte(versions))
+			os.Stdout.Write([]byte("master"))
+			os.Stdout.Write([]byte("\n"))
+		}
+	} else if program == "tarantool-ee" {
+		for _, v := range localFiles {
+			var versions []version.Version
+			if strings.Contains(v.Name(), "tarantool-enterprise-bundle") {
```

Comment:
Should we check that v is a file, not a directory or something else?

Reply:
added